doc/acl/SONiC_ACL_based_metering_hld.md

```diff
@@ -0,0 +1,430 @@
+                                           : next-hop ip address             Example: "10.0.0.1"
+                                           : next-hop group set of addresses Example: "10.0.0.1,10.0.0.3"
+
+mirror_action = 1*255VCHAR                 ; refer to the mirror session
+policer_action = 1*255VCHAR                ; refer to the policer name
+
+```
+### SWSS Enhancements
+- Query and save the POLCIER action capability in acl orchangent
+- As part of rule processing,
+    - Parse and validate policer info as part of rule action processing. Allow policer action only when capability is enabled.
+    - For platforms with action_type_list is mandatory on create, if rule is set with policer action, validation is added to check corresponding table action list and accept/drop the rule configuration with policer action.
+    - Set or disable policer action.
+- Policer cannot be deleted when referenced by ACL rule.
+
+### Configuration and management
+Note that policer configurations will leverage existing method.
+
+#### Config command
+
+In acl-loader script, an optional argument "--policer_name" will be taken as input and all rules configured as part of the json will be enabled with policer action.
+```
+acl-loader update full /tmp/test.json --policer_name policer_1
+```
+Alternatively, user can use "config load" command to configure the policer action or use sonic-cli interface.
+
+```c
+{
+   	"POLICER": {
+		"policer1": {
+		    "meter_type": "bytes",
+		    "mode": "sr_tcm",
+		    "cir": "12500",
+		    "cbs": "12500",
+		    "pir": "17500",
+		    "pbs": "17500",
+		    "color": "aware",
+		    "red_packet_action": "drop",
+		    "yellow_packet_action": "drop",
+		    "green_packet_action": "forward"
+        },
+        "policer2": {
+		    "meter_type": "bytes",
+		    "mode": "sr_tcm",
+		    "cir": "12500",
+		    "cbs": "12500",
+		    "pir": "17500",
+		    "pbs": "17500",
+		    "color": "aware",
+		    "red_packet_action": "drop",
+		    "yellow_packet_action": "drop",
+		    "green_packet_action": "forward"
+        }
+	},
+    "ACL_TABLE": {
+        "VRF_ACL_REDIRECT_V4": {
+            "policy_desc": "Redirect traffic to nexthop in different vrfs",
+            "type": "L3",
+            "ports": ["Ethernet1"]
+        },
+        "VRF_ACL_REDIRECT_V6": {
+            "policy_desc": "Redirect traffic to nexthop in different vrfs",
+            "type": "L3V6",
+            "ports": ["Ethernet1"]
+        }
+    },
+    "ACL_RULE": {
+        "VRF_ACL_REDIRECT_V4|rule1": {
+            "priority": "55",
+            "SRC_IP": "30.0.0.1",
+            "policer_action":"policer1"
+        },
+        "VRF_ACL_REDIRECT_V6|rule1": {
+            "priority": "55",
+            "SRC_IPV6": "2000:0030::1",
+            "policer_action":"policer2"
+        }
+    }
+}
+
+```
+
+#### Show command
+
+acl-loader script will be updated to display policer action in show output. 
+
+```c
+admin@sonic:~$ show acl rule
```

Comment:
How about policer statistics? Any plan to have CLI for ACL policer statistics? Please update the HLD.

Reply:
Since policer can be attached to different entities (port, mirror, acl), we will take it up as separate task. We will come up with separate HLD for the same. I will capture in Overview section that "policer statistics is not supported as part of this HLD" .